src/GitVersion.Core/VersionCalculation/VersionSearchStrategies/BaseVersionV2.cs

```diff
@@ -0,0 +1,87 @@
+namespace GitVersion.VersionCalculation;
+
+internal class BaseVersionV2 : BaseVersion
```

Comment:
Yes you are right on the long term the v1 needs to got to v2 as well. Because of the fact to have a clear focus on the new functionality I decided not to change any old logic and create a new type BaseVersionV2 with the new properties. Please notice that V2 extends V1.

Q1: Yes we can move the properties to BaseVersion but we need to change the logic in the other Strategies as well. Which migt be a good refactoring task which fitts into a separate PR.

Q2: From the API point of view you have different Strategies which are returning a list of BaseVersion's. We need to think of that it is just a temporary construct. Which will be refactored later. First of all you need to agree on the properties which are part of BaseVersionV2.

Reply:
thank you for clarifications. Can we add at the BaseVersionV2 level a TODO which states that this is a temp construct that should be merged into BaseVersion, so that we do not loose the reasoning (hopefully after we merge this PR you can address this in another PR and then the question is closed)

Reply:
Okay no problem I can do this. I have added a todo in code